packages/core-plugin-api/src/extendable-components/createExtendableComponent.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  createContext,
+  useContext,
+  useMemo,
+  ComponentType,
+  forwardRef,
+} from 'react';
+
+import { useComponentExtensions } from './ExtensionsProvider';
+import {
+  ExtendableComponentDescriptor,
+  ExtendableComponentRef,
+  ExtendableComponentRefConfig,
+  ExtendableComponentRefConfigForwardRef,
+  RefInfoProps,
+} from './types';
+
+export function _createExtendableComponent<Props extends {}, Context>(
+  config: ExtendableComponentRefConfig<Props, Context>,
+  useRef: boolean,
+): ExtendableComponentDescriptor<Props, Context> {
+  const componentRef = useRef
+    ? ({ ...config } as ExtendableComponentRef<Props, Context, RefInfoProps>)
+    : ({ ...config } as ExtendableComponentRef<Props, Context>);
+
+  const ctx = createContext<Context>(null as any);
```

Comment:
```suggestion
  const ctx = createContext<Context | undefined>(undefined);
```

And add a check that it's available when consuming, for safety :grin:

Similar to the ref concern above, this context isn't guaranteed to be unique for each extendable component, would need to work around that somehow too.